src/overlays/actors/ovl_En_Ru/z_en_ru.c

```diff
@@ -76,34 +86,292 @@ static DamageTable D_80A39488 = {
     /* Powder Keg     */ DMG_ENTRY(0, 0x0),
 };
 
-#endif
+static AnimationInfoS sAnimations[] = {
+    { &gAdultRutoIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gAdultRutoIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gAdultRutoRaisingArmsUpAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gAdultRutoCrossingArmsAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gAdultRutoLookingDownLeftAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gAdultRutoIdleHandsOnHipsAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gAdultRutoHeadTurnDownLeftAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gAdultRutoSwimmingUpAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+};
+
+// in PostLimbdraw, conversts limbIndex to bodyPartsPos index
+s8 sBodyPartPosIndexs[] = { -1, -1, 0x0C, 0x0D, 0x0E, -1, 0x09, 0x0A, 0x0B, -1,  0x00, 0x06,
+                            -1, -1, 0x07, 0x08, 0x02, -1, -1,   0x03, 0x4,  0x2, 0x1,  0 };
+
+s8 sRuBodyParts[] = {
+    0x00, 0x00, 0x00, 0x00, 0x03, 0x04, 0x00, 0x06, 0x07, 0x00, 0x09, 0x0A, 0x00, 0x0C, 0x0D, 0x00,
+};
+
+u8 sRuShadowSizes[] = {
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+static TrackOptionsSet sTrackOptions = {
+    { 0xFA0, 4, 1, 3 },
+    { 0x1770, 4, 1, 6 },
+    { 0xFA0, 4, 1, 3 },
+    { 0x1770, 4, 1, 6 },
+};
+
+Vec3f sHeadFocusPos = {
+    800.0f,
+    0,
+    0,
+};
+
+Vec3f sBodyFocusPos = {
+    0,
+    0,
+    0,
+};
+
+// unused, display lists? (not in oot ru2 either)
+// if they were display lists they would be
+//  gDPSetRenderMode(POLY_OPA_DISP++, AA_EN | Z_CMP | Z_UPD | IM_RD | CLR_ON_CVG | CVG_DST_WRAP | ZMODE_XLU | FORCE_BL |
+//  G_RM_FOG_SHADE_A, AA_EN | Z_CMP | Z_UPD | IM_RD | CLR_ON_CVG | CVG_DST_WRAP | ZMODE_XLU | FORCE_BL |
+//  GBL_c2(G_BL_CLR_IN, G_BL_A_IN, G_BL_CLR_MEM, G_BL_1MA));
+// gDPSetAlphaCompare(POLY_OPA_DISP++, G_AC_THRESHOLD);
+Gfx D_80A39598[] = {
+    0xE200001C,
+    0xC81049F8,
+    0xE2001E01,
+    0x00000001,
+};
+
+Gfx sRutoEmptyDisplayList[] = {
+    0xDF000000, 0x00000000 // gSPEndDisplayList();
+};
+
+s32 EnRu_ChangeAnimation(SkelAnime* skelAnime, s16 animIndex) {
+    s16 lastFrame;
+    s32 ret = false;
+
+    if ((animIndex >= 0) && (animIndex < 8)) { // cannot be sizeof() because slti vs sltiu
```

Comment:
So `ARRAY_COUNT` doesn't work?

Reply:
Oh yeah that works, forgot they were different